Userland/Applications/NetworkSettings/CertStoreDialog.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2023, Fabian Dellwing <fabian@dellwing.net>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Applications/NetworkSettings/CertStoreDialog.h>
+#include <Applications/NetworkSettings/CertStoreDialogGML.h>
+#include <LibCrypto/ASN1/PEM.h>
+
+namespace NetworkSettings {
+
+NonnullRefPtr<CertStoreModel> CertStoreModel::create() { return adopt_ref(*new CertStoreModel); }
+
+void CertStoreModel::load()
+{
+    // FIXME: In the future we will allow users to import own CAs, we than need to change this logic
+    auto cacert_result = Core::File::open("/etc/cacert.pem"sv, Core::File::OpenMode::Read);
+    if (cacert_result.is_error()) {
+        dbgln("Failed to load CA Certificates: {}", cacert_result.error());
+        return;
+    }
+    auto cacert_file = cacert_result.release_value();
+    auto data_result = cacert_file->read_until_eof();
+    if (data_result.is_error()) {
+        dbgln("Failed to load CA Certificates: {}", data_result.error());
+        return;
+    }
+    auto data = data_result.release_value();
+
+    auto load_result = DefaultRootCACertificates::the().reload_certificates(data);
+    if (load_result.is_error()) {
+        dbgln("Failed to load CA Certificates: {}", load_result.error());
+        return;
+    }
+
+    m_certificates = load_result.release_value();
+}
+
+DeprecatedString CertStoreModel::column_name(int column) const
+{
+    switch (column) {
+    case Column::Issued_To:
+        return "Issued To";
+    case Column::Issued_By:
+        return "Issued By";
+    case Column::Expire:
+        return "Expiration Date";
+    default:
+        VERIFY_NOT_REACHED();
+    }
+}
+
+GUI::Variant CertStoreModel::data(GUI::ModelIndex const& index, GUI::ModelRole role) const
+{
+    if (role != GUI::ModelRole::Display)
+        return {};
+    if (m_certificates.is_empty())
+        return {};
+
+    auto cert = m_certificates.at(index.row());
+
+    switch (index.column()) {
+    case Column::Issued_To:
+        return cert.subject.subject.is_empty() ? cert.subject.unit : cert.subject.subject;
+    case Column::Issued_By:
+        return cert.issuer.subject.is_empty() ? cert.issuer.unit : cert.issuer.subject;
+    case Column::Expire:
+        return cert.not_after.to_deprecated_string("%Y-%m-%d"sv);
+    default:
+        VERIFY_NOT_REACHED();
+    }
+
+    return {};
+}
+
+void CertStoreDialog::open_cert_store(Window* parent_window)
+{
+    auto dialog = CertStoreDialog::construct(parent_window);
+
+    dialog->exec();
+}
+
+CertStoreDialog::CertStoreDialog(Window* parent_window)
+    : Dialog(parent_window)
+{
+    auto widget = set_main_widget<GUI::Widget>().release_value_but_fixme_should_propagate_errors();
```

Comment:
Is it possible for us to use a `try_create` function instead? This is a new addition, I feel like we shouldn't be using `release_value_but_fixme_should_propagate_errors` usages in new code.

Reply:
That would be a big yak to shave as far as I can tell as a C++ noob. The whole NetworkSettings would need to be touched for that? Nothing there seems ready to handle errors..

Reply:
I could implement ErrorOr in the new code and MUST it away for the moment in the old code. Would that be ok?

Reply:
Ok, I got the `try_create` working, but without refactoring `NetworkSettings` there will be one `release_value_but_fixme_should_propagate_errors ` left :)